script/generatetypescriptfile.py

```diff
@@ -0,0 +1,166 @@
+import openspace
+import asyncio
+import re
+import shutil
+
+StaticFileName = "openspace-api-static.txt"
+outFileName = "../declaration/openspace-api-js.d.ts"
+
+def writeFunctionDocumentation(file, fun):
+    """Writes the documentation of a function to file."""
+    # Obs, some documentation includes '\\param' which messes with the indentation but it
+    # should be fine, the documentation for the function still works as expected.
+    file.write("  /**\n   * ")
+    file.write(fun["help"])
+    file.write("\n   */\n")
+
+def getArgument(argument):
+    """
+    Returns the parameter name and type of an argument converting them to TS syntax.
+    """
+    parameterName = argument["name"]
+    parameterType = argument["type"]
+
+    # If a parameter is optional its marked with a '?' after the parameter type.
+    # Typescript requires the '?' to denote an optional parameter on the name rather than
+    # the type
+    if "?" in parameterType:
+        parameterName += "?"
+        parameterType = re.sub(r'\?', '', parameterType)
+
+    # Some function like printDebug can take any number of parameter which is specified
+    # with the '*' so we translate these into a rest parameter
+    if not parameterName and parameterType == "*":
+        parameterName = "...args"
+        parameterType = "any[]"
+
+    return f"{parameterName}: {parameterType.lower()}"
+
+def writeFunctionArguments(file, fun):
+    """Writes all the function parameters and types to file."""
+    file.write("(")
+    arguments = fun["arguments"]
+    if arguments:
+        for argument in arguments[:-1]:
+            file.write(f"{getArgument(argument)}, ")
+
+        file.write(f"{getArgument(arguments[-1])}")
+    file.write(")")
+
+def writeFunctionReturnType(file, fun):
+    """Writes the function return type, converting it to TS syntax in special cases."""
+    functionReturnType = fun["returnType"].lower()
+
+    if not functionReturnType:
+        functionReturnType = "void"
+
+    # If the return type is a map
+    if "->" in functionReturnType:
+        types = functionReturnType.split("->")
+        firstType = types[0].strip()
+        secondType = types[1].strip()
+        functionReturnType = f"Map<{firstType}, {secondType}>"
+
+    # If we are returning a tuple of items
+    if "(" in functionReturnType and ")" in functionReturnType:
+        functionReturnType = re.sub('\\(', "[", functionReturnType)
+        functionReturnType = re.sub('\\)', "]", functionReturnType)
+
+    file.write(f" => Promise<{functionReturnType}>\n")
+
+def writeFunctionToFile(file, fun):
+    """
+    Writes a function with corresponding documentation, arguments, and return type to file
+    """
+    writeFunctionDocumentation(file, fun)
+    file.write(f'  {fun["name"]}: ')
```

Comment:
This is only allowed in Python 3.12 and forward. https://peps.python.org/pep-0701/
I think that is fine, but say 3.8 in the readme